packages/rest/src/validation/request-body.validator.ts

```diff
@@ -79,11 +56,23 @@ function convertToJsonSchema(openapiSchema: SchemaObject) {
   return jsonSchema;
 }
 
-// tslint:disable-next-line:no-any
-function validateValueAgainstJsonSchema(body: any, schema: any) {
-  const ajv = new AJV({allErrors: true});
+function validateValueAgainstJsonSchema(
+  // tslint:disable-next-line:no-any
+  body: any,
+  // tslint:disable-next-line:no-any
+  schema: any,
+  globalSchemas?: SchemasObject,
+) {
+  let schemaWithRef = _.cloneDeep(schema);
```

Comment:
Since we are changing the top-level `components` property only, is it really needed to perform a potentially expensive deep clone?

My proposal:

```ts
const schemaWithComponents = Object.assign({}, schema);
schemaWithComponents.components = {
  schemas: globalSchemas,
};
```